src/ImpactEvaluator.sol

```diff
@@ -86,69 +88,42 @@ contract ImpactEvaluator is AccessControl, Nonces {
             addresses.length == scores.length,
             "Addresses and scores length mismatch"
         );
-        require(roundIndex < currentRoundIndex, "Round not finished");
-
-        (uint indexInOpenRounds, Round storage round) = getOpenRound(
-            roundIndex
+        require(
+            previousRound.index != currentRound.index &&
+            roundIndex == previousRound.index,
+            "Can only score previous round"
         );
-        uint sumOfScores = validateScores(scores, round.totalScores);
-        reward(round, addresses, scores);
-        round.totalScores += sumOfScores;
+        require(!previousRound.scored, "Round already scored");
 
-        if (round.totalScores == MAX_SCORE) {
-            deleteOpenRound(indexInOpenRounds);
-        }
-    }
+        uint sumOfScores = validateScores(scores);
+        reward(addresses, scores);
+        previousRound.totalScores += sumOfScores;
 
-    function getOpenRound(
-        uint roundIndex
-    ) private view returns (uint index, Round storage) {
-        for (uint i = 0; i < openRounds.length; i++) {
-            if (openRounds[i].index == roundIndex) {
-                return (i, openRounds[i]);
-            }
+        if (previousRound.totalScores == MAX_SCORE) {
+            previousRound.scored = true;
```

Comment:
What is the benefit of storing `scored` in the on-chain state as opposed to calculating it on the fly? Isn't always true that `scored = totalScores >= MAX_SCORE`?

Reply:
I also find the name `scored` misleading - if we submit partial scores, then `scored` stays `false`, despite the fact that we did record some scores.

Proposals for a better name: `scoredAll` or `fullyScored`.

Reply:
Excellent suggestion! I removed `scored`